### PR TITLE
Code review

### PR DESCRIPTION
**Pull request by StuartCowley**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,70 +1,81 @@
...
+```bash
+npm test
+```
 
-### Deployment
+## If I had more time...
+I would add better alt text to the images in image results. Currently they all have the alt text of "placeholder" which is not useful at all. 
 
-This section has moved here: [https://facebook.github.io/create-react-app/docs/deployment](https://facebook.github.io/create-react-app/docs/deployment)
+I would need to process the data recieved from the API differently so that the alt text from NASA could be added to the individual image in the same way the href is added to the src attribute.
```
Comment (StuartCowley) [inline README.md:74]: Typo in received:
```
I would need to process the data received from the API differently so that the alt text from NASA could be added to the individual image in the same way the href is added to the src attribute.
```

File: public/index.html [hunk 2]
```diff
@@ -2,42 +2,20 @@
 <html lang="en">
   <head>
     <meta charset="utf-8" />
-    <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
+    <link rel="icon" href="%PUBLIC_URL%/nasa-favicon.png" />
```
Comment (StuartCowley) [inline public/index.html:5]: Nice touch!

File: src/__tests__/app.test.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { render, screen } from "@testing-library/react";
```
Comment (StuartCowley) [inline src/__tests__/app.test.js:1]: This file should written in PascalCase so it matches the `App.js` component name

File: src/__tests__/SearchResults.test.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+describe("SearchResults", () => {
+  let resultsMock = ["test1", "test2"];
+
+  it("renders results correctly", () => {
+    const { asFragment } = render(<SearchResults results={resultsMock} />);
+
+    expect(asFragment()).toMatchSnapshot();
+  });
+
```
Comment (StuartCowley) [inline src/__tests__/SearchResults.test.js:12]: Could another test be added here to assert the correct number of images are being rendered? Could do this with:

```
  it("renders correct number of results", () => {
    render(<SearchResults results={resultsMock} />);
    const images = screen.getAllByRole('img');

    expect(images).toHaveLength(2)
  });
```

There is a bit of a gotcha with this though... You will need to create the `resultsMock` at the top of the describe block and use a `beforeEach` hook to set the initial values used for most tests:
```
let resultsMock;

beforeEach(() => {
  resultsMock = ["test1", "test2"];
});
```
If you just add the new test without doing this extra setup, the order you put the tests in the file will determine whether or not they will all pass, as if you set `resultsMock` to empty in one test then the next one will also use that same value, meaning the component will render the 'no results found' message. Adding the `beforeEach` will reset the props value before each test so you have more fine grained control over what is passed to the component on each iteration through

File: src/__tests__/app.test.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+import App from "../components/App";
+
+describe("App", () => {
+  it("renders App correctly", () => {
+    const { asFragment } = render(<App />);
+    const nasaImg = screen.getByAltText("NASA in red stylised font");
+
+    expect(asFragment()).toMatchSnapshot();
+    expect(nasaImg).toHaveAttribute("src");
```
Comment (StuartCowley) [inline src/__tests__/app.test.js:10]: Did you know you can search not just for the attributes on an element, but also its value? EG:
```
expect(nasaImg).toHaveAttribute("src", 'https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg');
```
Makes for a more robust test if you can verify its exactly what you expect

File: src/components/SearchResults.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
+import React from "react";
+import PropTypes from "prop-types";
+import "../styles/search-results.css";
+
+function SearchResults({ results }) {
+  if (!results.length) {
+    return (
+      <p className="no-images-message">Sorry, we couldn't find anything.</p>
```
Comment (StuartCowley) [inline src/components/SearchResults.js:8]: Looking at the rest of the classnames here, could this also be named using the atomic style to `search-results__message` ?

File: src/components/SearchResults.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+            key={index}
+          />
+        );
+      })}
+    </div>
+  );
+}
+
+SearchResults.propTypes = {
+  results: PropTypes.array.isRequired,
```
Comment (StuartCowley) [inline src/components/SearchResults.js:28]: How about verifying what each item in the array will look like? Is it an array of objects/strings/numbers etc?

File: src/requests/getImages.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+  } else {
+    return axios
+      .get(`https://images-api.nasa.gov/search?q=${query}`)
+      .then((response) => {
+        const imageResults = response.data.collection.items;
+        const parsedImages = imageResults.filter(
+          (element) => element.data[0].media_type === "image"
+        );
+        const images = parsedImages.map((element) => element.links[0].href);
+        return images;
```
Comment (StuartCowley) [inline src/requests/getImages.js:15]: Minor point but you could condense this down a bit:
```
        return parsedImages.map((element) => element.links[0].href);
```

File: src/styles/search.css [hunk 9]
```diff
@@ -0,0 +1,26 @@
+.search-form {
+  display: flex;
+  flex-direction: row;
```
Comment (StuartCowley) [inline src/styles/search.css:3]: `flex-direction` is set to `row` implicitly, you dont need to declare it like this unless you are overwriting a previous definition. Can delete this no problem

**Review by StuartCowley — COMMENTED** [review]
Nice work! Have left a few suggestions on improving the tests, some minor formatting things, and a few other bits. Good to see the stretch goals in the readme too!
I couldnt put a comment on the files themselves but you should delete the `.DS_Store`, `public/.DS_Store`, `src/.DS_Store` and `images/.DS_Store` files, as they have no use to others

Good stuff here, keep it up!
